armi/reactor/blueprints/__init__.py

```diff
@@ -329,21 +335,23 @@ def _prepConstruction(self, cs):
             runLog.header("=========== Verifying Assembly Configurations ===========")
             self._checkAssemblyAreaConsistency(cs)
 
-            if not cs["detailedAxialExpansion"]:
+            if not cs[CONF_DETAILED_AXIAL_EXPANSION]:
                 # this is required to set up assemblies so they know how to snap
                 # to the reference mesh. They wont know the mesh to conform to
                 # otherwise....
                 axialExpansionChanger.makeAssemsAbleToSnapToUniformMesh(
-                    self.assemblies.values(), cs["nonUniformAssemFlags"]
+                    self.assemblies.values(), cs[CONF_NON_UNIFORM_ASSEM_FLAGS]
                 )
-            if not cs["inputHeightsConsideredHot"]:
+            if not cs[CONF_INPUT_HEIGHTS_HOT]:
                 runLog.header(
                     "=========== Axially expanding all assemblies (except control) from Tinput to Thot ==========="
                 )
                 # expand axial heights from cold to hot so dims and masses are consistent
                 # with specified component hot temperatures.
                 axialExpansionChanger.expandColdDimsToHot(
-                    self.assemblies.values(), cs["detailedAxialExpansion"]
+                    self.assemblies.values(),
+                    cs[CONF_DETAILED_AXIAL_EXPANSION],
+                    cs[CONF_ASSEM_FLAGS_SKIP_AXIAL_EXP],
```

Comment:
This is not incredibly important, but...

With this change, you change the API of `expandCoddDimsToHot()`. And in fact you make the new parameter mandatory, so the change is also not backwards compatible.

My inclination would be to avoid changing the signature of `expandColdDimsToHot()`, and instead to just whittle down the list returned from `self.assemblies.values()` right here when you pass it in. Something like the following:

```python
assemsToExpand = [a for a in list(self.assemblies.values()) if not a.hasFlags(cs[CONF_ASSEM_FLAGS_SKIP_AXIAL_EXP])
axialExpansionChanger.expandColdDimsToHot(assemsToExpand, cs[CONF_DETAILED_AXIAL_EXPANSION])
```

Just a suggestion.

Reply:
Addressed. https://github.com/terrapower/armi/pull/1235/commits/32ce614201e443c1a4e6db86c634f16c0f8856ad